packages/amplify_core/lib/src/category/amplify_storage_category.dart

```diff
@@ -20,11 +20,13 @@ class StorageCategory extends AmplifyCategory<StoragePluginInterface> {
   @nonVirtual
   Category get category => Category.storage;
 
-  Future<UploadFileResult> uploadFile(
-      {required File local,
-      required String key,
-      void Function(TransferProgress)? onProgress,
-      UploadFileOptions? options}) {
+  Future<UploadFileResult> uploadFile({
+    // TODO(HuiSF): AWSFile
+    required dynamic /* File */ local,
```

Comment:
*nit:* Does this need to change now? This isn't a dep on Flutter, and I don't think the import from dart:io should be an issue.